addon/components/sl-date-time.js

```diff
@@ -122,23 +128,28 @@ export default Ember.Component.extend( TooltipEnabled, {
         'momentValue',
         function() {
             const momentValue = this.get( 'momentValue' );
-            let formattedString = '';
-
-            switch ( this.get( 'format' ) ) {
-                case Format.DATE:
-                    formattedString = momentValue.format( 'YYYY-MM-DD' );
-                    break;
-
-                case Format.RELATIVE:
-                    formattedString = momentValue.fromNow();
-                    break;
-
-                default:
-                case Format.DATETIME:
-                    formattedString =
-                        momentValue.format( 'dddd, MMMM Do YYYY, h:mm A' ) +
-                        ' ' +
-                        this.get( 'timezoneString', 'en' );
+            let formattedString = null;
+
+            if ( momentValue ) {
+                switch ( this.get( 'format' ) ) {
+                    case Format.DATE:
+                        formattedString = momentValue.format( 'YYYY-MM-DD' );
+                        break;
+
+                    case Format.RELATIVE:
+                        formattedString = momentValue.fromNow();
+                        break;
+
+                    default:
+                    case Format.DATETIME:
+                        if ( this.get( 'timezoneString' ) ) {
+                            formattedString =
+                                momentValue.format( 'dddd, MMMM Do YYYY, h:mm A' ) +
+                                ' ' +
+                                this.get( 'timezoneString', 'en' );
+                        }
+                        break;
+                }
```

Comment:
It is acceptable for this computed property to return an empty string value, so all of these changes can be reverted.